graphite_beacon/handlers/telegram.py

```diff
@@ -23,89 +44,173 @@ def init_handler(self):
+                    LOGGER.debug(
+                            'Deleting chat [%s] from notify list.', chat_id)
+                    reply_text = 'Deactivated.'
+                    chats.remove(chat_id)
+                    if chatfile:
+                        write_to_file(chats, chatfile)
+
+            elif command == '/help':
+                reply_text = HELP_MESSAGE
+
             else:
-                continue
+                LOGGER.warning(BOT_COMMAND_ERRORS[command])
+
+            if reply_text:
+                yield self.sendMessage({
+                        'chat_id': chat_id,
+                        'reply_to_message_id': message_id,
+                        'text': reply_text,})
 
     @gen.coroutine
     def notify(self, level, *args, **kwargs):
 
-        LOGGER.debug("Handler (%s) %s", self.name, level)
+        LOGGER.debug('Handler (%s) %s', self.name, level)
+
+        notify_text = self.get_message(level, *args, **kwargs)
+        for chat in self.chats:
+            yield self.sendMessage({"chat_id": chat, "text": notify_text})
 
-        message = self.get_message(level, *args, **kwargs)
-        for chat in self._chats:
-            yield self.client.fetch(
-                self.url + "sendMessage", body=json.dumps({"chat_id": chat, "text": message}),
-                method="POST", headers={"Content-Type": "application/json"})
+    def get_message(self, level, alert, value,
+                    target=None, ntype=None, rule=None):
+
+        msg_type = 'telegram' if ntype == 'graphite' else 'short'
+        tmpl = TEMPLATES[ntype][msg_type]
+        generated = tmpl.generate(
+            level=level, reactor=self.reactor, alert=alert,
+            value=value, target=target,)
+        return generated.decode().strip()
+
+def write_to_file(chats, chatfile):
+    with open(chatfile, 'w') as fh:
+        chats = map(str, chats)
+        fh.write('\n'.join(chats))
+
+def get_chatlist(chatfile):
+    if not chatfile:
+        return set()
+    try:
+        with open(chatfile) as fh:
+            return set(int(chat) for chat in fh)
+    except Exception as e:
+        LOGGER.error('could not load saved chats:\n%s' % (e))
+        return set()
+
+def get_data(upd, bot_ident):
+    update_content = json.loads(upd.decode())
+    result = update_content['result']
+    data = (get_fields(update, bot_ident) for update in result)
+    return filter(None, data)
+
+def get_fields(upd, bot_ident):
+    """In telegram api, not every update has message field,
+    and not every message has update field.
+    We skip those cases. Rest of fields are mandatory.
+    We also skip if text is not a valid command to handler.
+    """
+    msg = upd.get('message', {})
+    text = msg.get('text')
+    if not text:
+        return
+    chat_id  = msg['chat']['id']
+    command = filter_commands(text, chat_id, bot_ident)
+    if not command:
+        return
+    return (upd['update_id'], chat_id, msg['message_id'], command)
+
```

Comment:
Do you mind running pylint on this file? There are some small things like 2 newlines between functions that would be nice to see :)